app/repositories/app_usage_event_repository.rb

```diff
@@ -145,7 +145,7 @@ def purge_and_reseed_started_apps!
       end
 
       def delete_events_older_than(cutoff_age_in_days)
-        Database::OldRecordCleanup.new(AppUsageEvent, cutoff_age_in_days, keep_at_least_one_record: true).delete
+        Database::OldRecordCleanup.new(AppUsageEvent, cutoff_age_in_days, keep_at_least_one_record: true, keep_running_app_records: true).delete
```

Comment:
Have y'all done any performance testing around the impact this might have on Cloud Controllers on foundations with a large number of running apps? I think even with the thirty-day expiry for events large environments can get pretty bogged-down.

@zrob might have some more historical context here.

Reply:
I see there were some previous stories around performance issues while deleting old records, and I'll investigate the impact my PR might have. Thanks for sharing that context.